packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -0,0 +1,158 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useQuery, useLazyQuery, useMutation } from '@apollo/react-hooks';
+
+import { useCartContext } from '@magento/peregrine/lib/context/cart';
+
+export const displayStates = {
+    DONE: 'done',
+    EDITING: 'editing'
+};
+
+export const useShippingMethod = props => {
+    const {
+        onSave,
+        queries: { getShippingMethods, getSelectedShippingMethod },
+        mutations: { setShippingMethod }
+    } = props;
+
+    const [{ cartId }] = useCartContext();
+
+    /*
+     *  Apollo Hooks.
+     */
+    const {
+        data: chosenShippingMethodData,
+        loading: isLoadingSelectedShippingMethod
+    } = useQuery(getSelectedShippingMethod);
+    const [
+        fetchShippingMethods,
+        { data, error, loading: isLoadingShippingMethods }
+    ] = useLazyQuery(getShippingMethods);
```

Comment:
Also don't forget your fetch policy for these.

Reply:
Thanks - I was thinking I wanted the default `cache-first`,  but folks could update shipping addresses and selected methods in other tabs, etc. so I went with `cache-and-network`: https://github.com/magento/pwa-studio/pull/2280/commits/2ad41c7d66af0e1b2a6815d69846c9512452aee7